### ISSUE TITLE
Resource missing for web app config

### ISSUE THREAD

**Issue by WillooWisp** [pulumi/pulumi-azure-native#651]
The problem I have right now is that I am unable to correctly sync state by import, refresh and up, since the resource for web app config is missing.

Below is the API for the missing resource. It seems to contain the missing properties of the SiteConfig from the WebApp resource so together with the WebAppApplicationSettings and WebAppConnectionStrings resources a web app can be managed completely by pulumi.

`GET / PUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/web?api-version=2019-08-01
`
```
{
"type": "Microsoft.Web/sites/config",
    "apiVersion": "2018-11-01",
    "name": "[concat(parameters('sites_name'), '/web')]",
    "location": "North Europe",
    "dependsOn": [
        "[resourceId('Microsoft.Web/sites', parameters('sites_name'))]"
    ],
    "properties": {
        "numberOfWorkers": -1,
        "defaultDocuments": [
            "Default.htm",
            "Default.html",
            "Default.asp",
            "index.htm",
            "index.html",
            "iisstart.htm",
            "default.aspx",
            "index.php"
        ],
        "netFrameworkVersion": "v4.0"
    }
}
```

**Comment by WillooWisp** [comment]
For my own planning of adoption to pulumi it would be nice to know if this will be fixed and it what release and when that can be expected? I was under the impression that one of the big benefits with the new azure native provider was that all resources would be supported more or less automatically without delays due to the "auto-generation" of resources?

**Comment by mikhailshilkov** [comment]
The `Web` provider has a long history and its API is odd in many ways. I [opened a PR](https://github.com/pulumi/pulumi-azure-native/pull/660) for this - it's basically one line of code for us - but the PUT endpoint doesn't really as it should, see https://github.com/Azure/azure-rest-api-specs/issues/13401

**Comment by mikhailshilkov** [comment]
@WillooWisp By the way - which properties are you missing from the parent `WebApp` resource? The `WebAppConfiguration` resource seems to have exactly the same properties as `WebApp.SiteConfig`?

**Comment by WillooWisp** [comment]
Yes but I hoped this separate config resource could make up for the problem where the web app api you use fpr the web app resource does not give out the site config, it is null.

Reported by you from one of my other issue reported related to the SiteConfig. 
https://github.com/Azure/azure-rest-api-specs/issues/12104

**Comment by mikhailshilkov** [comment]
Ah, okay, that's subtle. But yes, it seems to work in this resource from what I've seen.

**Comment by WillooWisp** [comment]
Any plans from MS side to fix the put issue?

**Comment by thomas11** [comment]
Reading the SiteConfig is now implemented by #3464.
